internal/test/helpers.go

```diff
@@ -47,10 +48,28 @@ func OKs(tb testing.TB, errs []error) {
 }
 
 // Equals fails the test if exp is not equal to act.
-func Equals(tb testing.TB, exp, act interface{}) {
+// msg is optional message to be printed, first param being format string and rest being arguments.
+func Equals(tb testing.TB, exp, act interface{}, msgs ...string) {
 	tb.Helper()
 	if !reflect.DeepEqual(exp, act) {
-		tb.Fatalf("\033[31m\n\n\texp: %#v\n\n\tgot: %#v\033[39m\n\n", exp, act)
+		var msgString string
+		if len(msgs) == 0 {
+			tb.Fatalf("\033[31m\n\n\texp: %#v\n\n\tgot: %#v\033[39m\n\n", exp, act)
+		} else if len(msgs) == 1 {
+			tb.Fatalf("\033[31m\n\n\t"+msgs[0]+"\n\n\texp: %#v\n\n\tgot: %#v\033[39m\n\n", exp, act)
+		} else {
+			args := make([]interface{}, len(msgs)-1)
```

Comment:
you can replace the whole loop with `args := msgs[1:]`, and then inline it into `fmt.Sprintf`.

Reply:
That actually results in - "cannot use msgs[1:] (value of type []string) as []any value in argument to fmt.Sprintf"
Not sure if the loop can be avoided.

However, I can still clean it up a bit - 

args := make([]interface{}, len(msgs)-1)
for i, msg := range msgs[1:] {
     args[i] = msg
}
msgString = fmt.Sprintf(msgs[0], args...)

Reply:
Or I could create a separate Equalsf -

// Equals fails the test if exp is not equal to act.
func Equals(tb testing.TB, exp, act interface{}) {
	Equalsf(tb, exp, act, "")
}

// Equalsf fails the test if exp is not equal to act.
// msg is optional message with arguments to be printed.
func Equalsf(tb testing.TB, exp, act interface{}, msg string, args ...any) {
	tb.Helper()
	if !reflect.DeepEqual(exp, act) {
		tb.Fatalf("\033[31m\n\n\t"+fmt.Sprintf(msg, args...)+"\n\n\texp: %#v\n\n\tgot: %#v\033[39m\n\n", exp, act)
	}
}

Reply:
If think a single function would be more convenient.

Reply:
Updated to single function